internal/pkg/machine/controllers/machine_status.go

```diff
@@ -94,22 +95,22 @@ func (ctrl *MachineStatusController) Run(ctx context.Context, r controller.Runti
 			return err
 		}
 
-		config, err := safe.ReaderGetByID[*config.MachineConfig](ctx, r, config.V1Alpha1ID)
-		if err != nil {
-			if state.IsNotFoundError(err) {
-				if err = safe.WriterModify(ctx, r, runtime.NewMachineStatus(), func(res *runtime.MachineStatus) error {
-					res.TypedSpec().Stage = runtime.MachineStageMaintenance
-					res.TypedSpec().Status.Ready = true
+		config, err := machineconfig.GetComplete(ctx, r)
+		if err != nil && !state.IsNotFoundError(err) {
+			return err
+		}
 
-					return nil
-				}); err != nil {
-					return err
-				}
+		if config == nil || config.Provider().Machine() == nil {
```

Comment:
I guess this `config.Provider().Machine() == nil` check isn't necessary here anymore.

Reply:
good points, I found couple of another places as well.